### ISSUE TITLE
NaN or Inf found in input tensor.

### ISSUE THREAD

**Issue by Dingding250** [HonglinChu/SiamTrackers#155]
琳琳师兄，您好！
我在重新训练NanoTrackV3时，没有使用您给的nanotrackv3.pth作为预训练权重，而是使用Mobilenetv3_samll_1.0.pth作为预训练权重重新训练V3，但是一直出现下面的问题：
[2024-05-10 16:58:15,826-rk0-distributed.py<reduce_gradients>#135] gradients method is sum
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
...
[2024-05-10 16:58:16,873-rk0-train.py#238] Epoch: [1][20/6250] lr: 0.001000
        batch_time: 0.054689 (0.198604) data_time: 0.000239 (0.061547)
        cls_loss: nan (nan)
        loc_loss: nan (nan)     total_loss: nan (nan)

损失也全是0，我调整了学习率和batchsize也不行，我的数据集没问题，可以正常对NanoV1，V2进行重新训练。

**Comment by dqs932** [comment]
还有一个问题是，使用nanotrackv3.pth作为预训练权重进行训练时，会正常训练，打印损失。

我改了网络结构，仍然想调用nanotrackv3.pth作为预训练权重进行训练时，会显示部分权值不匹配，这是正常的，但是训练会中断，不应该是忽略这些权重，重新对他们训练吗？
model.load_state_dict(pretrained_dict, strict=False)设置成了False，按道理应该是会忽略这些不匹配的权重，正常训练的。
不知道为啥会中断。

![image](https://github.com/HonglinChu/SiamTrackers/assets/100916543/353a5fee-443c-4f54-a341-122367a97735)

**Comment by Dingding250** [comment]
已解决。

**Comment by CV-poo** [comment]
> Author

请问您这边怎么解决的Nan问题，我也遇到了相同的问题，恳请您的答复谢谢

**Comment by CV-poo** [comment]
@dqs932 请问您这边怎么解决的Nan问题，我也遇到了相同的问题，恳请您的答复谢谢

**Comment by ogvalt** [comment]
@dqs932 I'd like to know too, how did you solve nan problem?

**Comment by CV-poo** [comment]
> @dqs932 I'd like to know too, how did you solve nan problem?

I think the #95 can solve your problem。

**Comment by ogvalt** [comment]
> > @dqs932 I'd like to know too, how did you solve nan problem?
> 
> 
> 
> I think the #95 can solve your problem。

yes, adding proper initialisation worked for me.
thanks

**Comment by XY-ZXY** [comment]
> 还有一个问题是，使用nanotrackv3.pth作为预训练权重进行训练时，会正常训练，打印损失。
> 
> 我改了网络结构，仍然想调用nanotrackv3.pth作为预训练权重进行训练时，会显示部分权值不匹配，这是正常的，但是训练会中断，不应该是忽略这些权重，重新对他们训练吗？ model.load_state_dict(pretrained_dict, strict=False)设置成了False，按道理应该是会忽略这些不匹配的权重，正常训练的。 不知道为啥会中断。
> 
> ![image](https://github.com/HonglinChu/SiamTrackers/assets/100916543/353a5fee-443c-4f54-a341-122367a97735)

我想请问下权值不匹配是如何解决的呢，我在测试v3的模型时也出现这种问题
